Add 3D labeling/mesh use case

elevans opened the pull request:
This PR adds the **3D labeling and mesh analysis** use case to the RTD docs. Briefly, this script uses only ImgLib2 and SciJava Ops to segment only puncta within the nuclear space, creates meshes for each puncta and nucleus, and performs geometry measurements. The outputs are the ImgLabeling index image of the nuclear puncta and two results tables for each channel respectively. I attempted to create the script in an "agnostic" way where a user could steal the useful functions and/or modify the script with their own image processing needs and still have the infrastructure to get the tables/results out.

gselzer commented on docs/ops/doc/examples/labeling_mesh.rst:
```diff
@@ -0,0 +1,244 @@
+
+.. figure:: https://media.scijava.org/scijava-ops/1.0.1/labeling_mesh_example_dialog.png
+
+Once the script has been configured click **OK** to start the analysis. The script will display two tables (one for each channel with the given channel name) and the labeling
```
This took longer than I expected to run - might not be a bad idea to give the users an idea of how long it will take, and to let them know they can see which Op is running via the `TaskService` (unfortunately, I don't think there's an imagej.net wiki page on the `TaskService`, is there @ctrueden?)

Additionally, some representation of the expected table outputs would be nice.

elevans replied:
I added a new section on the expected outputs (i.e. the index image and the tables) in 7836ba5fa622365b7dc7f2432ae5049fa5f1aad3.

elevans replied:
I've already removed the fill holes operation so this script is much faster now. Mark said he was able to get it down to under 30 seconds when he trimmed the data to only a few planes.

gselzer reviewed (commented):
This page is very good @elevans, this will be a great addition to the docs.

Could small suggestions!

hinerm reviewed (changes requested):
This script took 4.5 minutes to run on my computer. I agree with the need to tell users to expect it to be long.

The bulk of the time for me was spent in:
1. `FloodFill`
2. marching cubes
3. `Opening` actions.

The only part that stood out as potential framework responsibility was ~20 seconds spent in `Progress.pingListeners` during marching cubes, via `AbstractRichOp.[pre/post]process`

I don't know how long to expect these operations to take for this size image. I do wonder, for the sake of the demo, if cutting down the number of planes to 5-10 or so would be reasonable?

FWIW, I tried with 5 slices and it took ~30 seconds.

elevans commented on docs/ops/doc/examples/labeling_mesh.rst:
```diff
@@ -0,0 +1,244 @@
+        ch_a_img = gaussian_subtraction(ch_a_img, 8.0)
+        ch_a_ths = ops.op("create.img").input(ch_a_img, BitType()).apply()
+        ops.op("threshold.triangle").input(ch_a_img).output(ch_a_ths).compute()
+        ch_a_mask = ops.op("morphology.open").input(ch_a_ths, HyperSphereShape(1), 4).apply()
```
I think seeing the neighborhood will make this easier to explain. A `HyperSphereShape` of radius 1 on a 2D plane looks like this:

![sphere_radius_2d](https://github.com/user-attachments/assets/2dfcb19e-ed73-4111-8e6b-a25398ad5e24)

The pattern is essentially a four connected structuring element. You're correct that the neighborhoods are _n_-dimensional. Here's the same `HyperShpereShape` of radius 1 in 3D:

![radius 3d](https://github.com/user-attachments/assets/cc772e94-f1ea-4a9a-9aab-27c786d7c285)

In this case we want to consider the XYZ data per channel, so a `HyperSphereShape`.

We do want to fill holes on the first channel, the puncta channel, because my segmentation strategy successfully extracts the puncta (Gaussian blur subtraction) and they are generally whole. We _ideally_ fill holes on the nuclear channel because the variable signal in the nucleus can produce internal regions that fall below threshold values. In this case, I wanted a whole nucleus to mask over the puncta in the logical operation steps, thus filling holes makes sense here. Ultimately however, its not needed. The threshold masks are good enough for this dataset and the fill holes Op is very slow.

elevans commented on docs/ops/doc/examples/labeling_mesh.rst:
```diff
@@ -0,0 +1,244 @@
+            """
+            # find C and Z axis indicies
+            c_idx = find_axis_index(image, "Channel")
+            z_idx = find_axis_index(image, "Z")
```
Fixed in 457dd94d95528703b96e3ea17688d588b00a2ac2.

elevans commented on docs/ops/doc/examples/labeling_mesh.rst:
```diff
@@ -0,0 +1,244 @@
+        #@ ImgPlus img
+        #@ String (visibility = MESSAGE, value = "<b>Channel settings</b>", required = false) ch_msg
+        #@ String (label = "Channel A name", value = "a") ch_a_name
+        #@ String (label = "Channel B name", value = "b") ch_b_name
```
Agreed! Fixed in 457dd94d95528703b96e3ea17688d588b00a2ac2.

elevans commented:
Thanks @gselzer and @hinerm for the review! I've addressed all your comments and incorporated essentially all the suggestions you made to speed up the script and improve the document.

The last speed up we can do is to trim the data down. We can:

1. Crop the data in the script using my the `transform.intervalView` Op .
2. Crop the source data on the server and just serve a smaller image to everyone.

Thoughts?

Also if there is time I'd like to add smaller script leveraging what Curtis did to get meshes visible in the viewer.

gselzer commented:
> The last speed up we can do is to trim the data down. We can:
> 
>     1. Crop the data in the script using the `transform.intervalView` Op .
> 
>     2. Crop the source data on the server and just serve a smaller image to everyone.

I'd highly prefer (1), as it would allow people to run the workflow on the full dataset if they comment out the relevant lines.

elevans commented:
I have added image cropping to the workflow and got about 20 seconds on my computer.
